patches/tModLoader/Terraria/ID/NPCID.TML.cs

```diff
@@ -55,5 +59,48 @@ public static partial class Sets
 		/// Note: This set DOES NOT DO ANYTHING if your NPC doesn't use the Vanilla TownNPC aiStyle (aiStyle == 7).
 		/// </remarks>
 		public static bool[] AllowDoorInteraction = Factory.CreateBoolSet();
+
+		/// <summary>
+		/// If <see langword="true"/>, this NPC type (<see cref="NPC.type"/>) will be immune to all debuffs and "tag" buffs by default.<br/><br/>
+		/// Use this for special NPC that cannot be hit at all, such as fairy critters, container NPCs like Martian Saucer and Pirate Ship, bound town slimes, and Blazing Wheel. Dungeon Guardian also is in this set to prevent the bonus damage from "tag" buffs.<br/><br/>
```

Comment:
```suggestion
		/// Use this for special NPCs that cannot be hit at all, such as fairy critters, container NPCs like Martian Saucer and Pirate Ship, bound town slimes, and Blazing Wheel. Dungeon Guardian also is in this set to prevent the bonus damage from "tag" buffs.<br/><br/>
```